src/sendqueue.rs

```diff
@@ -0,0 +1,94 @@
+use std::ptr::NonNull;
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use libc::c_uint;
+
+use crate::raw;
+use crate::Error;
+use crate::{Active, Capture};
+
+pub struct SendQueue {
+    squeue: NonNull<raw::pcap_send_queue>,
+    sync: bool,
+}
+
+impl SendQueue {
+    pub fn new(memsize: c_uint) -> Result<Self, Error> {
+        let squeue = unsafe { raw::pcap_sendqueue_alloc(memsize) };
+        let squeue = NonNull::new(squeue).ok_or(Error::InsufficientMemory)?;
+
+        Ok(Self {
+            squeue,
+            sync: false,
+        })
+    }
+
+    pub fn maxlen(&self) -> c_uint {
+        unsafe { (*self.squeue.as_ptr()).maxlen }
+    }
+
+    pub fn len(&self) -> c_uint {
+        unsafe { (*self.squeue.as_ptr()).len }
+    }
+
+    pub fn sync(&mut self, sync: bool) {
+        self.sync = sync;
+    }
+
+    /// Add a packet to the queue.
+    pub fn queue(&mut self, buf: &[u8]) -> Result<(), Error> {
+        let start = SystemTime::now();
+        let since_the_epoch = start
+            .duration_since(UNIX_EPOCH)
+            .expect("Time went backwards");
+
+        let s = since_the_epoch.as_secs();
+        let us = since_the_epoch.subsec_micros();
+
+        let pkthdr = raw::pcap_pkthdr {
+            ts: libc::timeval {
+                tv_sec: s as i32,
+                tv_usec: us as i32,
+            },
+            caplen: buf.len() as u32,
+            len: buf.len() as u32,
```

Comment:
cast are wrong here just remove them

Reply:
I've added a try_into, since the `len()` is `usize` which is potentially longer than the `u32`.

Reply:
ah yeah indeed much better